espnet2/tasks/enh_s2t.py

```diff
@@ -0,0 +1,465 @@
+
+    @classmethod
+    def build_preprocess_fn(
+        cls, args: argparse.Namespace, train: bool
+    ) -> Optional[Callable[[str, Dict[str, np.array]], Dict[str, np.ndarray]]]:
+        assert check_argument_types()
+        if args.use_preprocessor:
+            if "st" in args.subtask_series:
+                retval = MutliTokenizerCommonPreprocessor(
+                    train=train,
+                    token_type=[args.token_type, args.src_token_type],
+                    token_list=[args.token_list, args.src_token_list],
+                    bpemodel=[args.bpemodel, args.src_bpemodel],
+                    non_linguistic_symbols=args.non_linguistic_symbols,
+                    text_cleaner=args.cleaner,
+                    g2p_type=args.g2p,
+                    # NOTE(kamo): Check attribute existence for backward compatibility
+                    rir_scp=args.rir_scp if hasattr(args, "rir_scp") else None,
+                    rir_apply_prob=args.rir_apply_prob
+                    if hasattr(args, "rir_apply_prob")
+                    else 1.0,
+                    noise_scp=args.noise_scp if hasattr(args, "noise_scp") else None,
+                    noise_apply_prob=args.noise_apply_prob
+                    if hasattr(args, "noise_apply_prob")
+                    else 1.0,
+                    noise_db_range=args.noise_db_range
+                    if hasattr(args, "noise_db_range")
+                    else "13_15",
+                    speech_volume_normalize=args.speech_volume_normalize
+                    if hasattr(args, "speech_volume_normalize")
+                    else None,
+                    speech_name="speech",
+                    text_name=["text", "src_text"],
+                )
+            else:
+                retval = CommonPreprocessor_multi(
+                    train=train,
+                    token_type=args.token_type,
+                    token_list=args.token_list,
+                    bpemodel=args.bpemodel,
+                    non_linguistic_symbols=args.non_linguistic_symbols,
+                    text_name=["text"],
+                    text_cleaner=args.cleaner,
+                    g2p_type=args.g2p,
+                )
+        else:
+            retval = None
+        assert check_return_type(retval)
+        return retval
+
+    @classmethod
+    def required_data_names(
+        cls, train: bool = True, inference: bool = False
+    ) -> Tuple[str, ...]:
+        if not inference:
+            retval = ("speech", "speech_ref1", "text")
+        else:
+            # Recognition mode
+            retval = ("speech",)
+        return retval
+
+    @classmethod
+    def optional_data_names(
+        cls, train: bool = True, inference: bool = False
+    ) -> Tuple[str, ...]:
+        retval = ["dereverb_ref"]
+        retval += ["speech_ref{}".format(n) for n in range(2, MAX_REFERENCE_NUM + 1)]
+        retval += ["text_ref{}".format(n) for n in range(2, MAX_REFERENCE_NUM + 1)]
```

Comment:
Maybe remove this line, because`text_ref*` seems not prepared in `egs2/TEMPLATE/enh_asr1/enh_asr.sh` and `egs2/TEMPLATE/enh_st1/enh_st.sh`

Reply:
OK